docs/usage/configuration/config_documentation.md

```diff
@@ -484,7 +484,8 @@ Sub-options for each listener include:
 
    * `names`: a list of names of HTTP resources. See below for a list of valid resource names.
 
-   * `compress`: set to true to enable HTTP compression for this resource.
+   * `compress`: set to true to enable gzip compression on HTTP bodies for this resource. This is currently only supported with the
+     `client`, `consent` and `metrics` resources.
```

Comment:
Wait, do we really not do this for federation endpoints?

Reply:
Yup, we just ignore `compress` if it's set for this resource:

https://github.com/matrix-org/synapse/blob/f30bcbd84a651de59777b2a749850f6ca56ce3f0/synapse/app/homeserver.py#L222-L223

I didn't open a new issue initially because I thought https://github.com/matrix-org/synapse/issues/2434 was about this but now that I'm reading this issue again it seems to assume it's possible, which doesn't show in the code.